app/assets/javascripts/Components/Modals/submission_file_upload_modal.jsx

```diff
@@ -16,18 +17,69 @@ class SubmissionFileUploadModal extends React.Component {
 
   onSubmit = event => {
     event.preventDefault();
-    this.props.onSubmit(this.state.newFiles, this.state.unzip);
+    if (this.state.newFiles.length == 1) {
+      this.props.onSubmit(this.state.newFiles, this.state.unzip, this.state.single_file_name);
+    } else {
+      this.props.onSubmit(this.state.newFiles, this.state.unzip);
+    }
   };
 
   handleFileUpload = event => {
-    this.setState({newFiles: event.target.files});
+    this.setState({newFiles: event.target.files, single_file_name: event.target.files[0].name});
   };
 
   toggleUnzip = () => {
     this.setState({unzip: !this.state.unzip});
   };
 
+  handleNameChange = event => {
+    this.setState({single_file_name: event.target.value});
+  };
+
   render() {
+    let fileInput;
+    if (this.state.newFiles.length == 1) {
```

Comment:
In Javascript you should generally use `===` instead of `==` for equality checking (the former is stricter than the latter).

See e.g. https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons